addon/instance-initializers/content-security-policy.js

```diff
@@ -0,0 +1,34 @@
+import { assert } from '@ember/debug';
+
+// reads addon config stored in meta element
+function readAddonConfig(appInstance) {
+  let config = appInstance.resolveRegistration('config:environment');
```

Comment:
Doesn't this read the config from the "legacy" location (in `config/environment.js`)?

Reply:
You are right that it reads the run-time configuration, which is normally provided by consumer via `config/environment.js`. But in this case the configuration is meant to be provided by addon's `config` hook only. It's a subset of addons configuration. It only includes `reportOnly` option and the build policy string. Both are required at run-time for FastBoot support.